doc/user/content/sql/idioms.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "Idioms"
+description: "Recommended SQL idioms."
+weight: 12
+menu:
+  main:
+    parent: sql
+---
+
+Some common transformations are surprisingly difficult to express in SQL. This
+page documents our recommended idioms for several of these hard-to-express
+transformations.
+
+These idioms are known to the Materialize optimizer, so using these idioms can
+result in much better performance than using idioms from other RDBMSes.
+
+## Top-K by group
+
+Suppose you want to group rows in a table by some key, then filter out all but
+the first _k_ elements within each group according to some ordering. In other
+databases you might use window functions. In Materialize, we recommend using a
+[`LATERAL` subquery](/sql/join/#lateral-subqueries). The general form of the
+query looks like this:
+
+```sql
+SELECT * FROM
+    (SELECT DISTINCT key_col FROM tbl) grp,
+    LATERAL (
+        SELECT col1, col2... FROM <tbl>
```

Comment:
not sure if `tbl` and `<tbl>` are meant to signify that they're different in some way?

Reply:
Oops, nope. I was going back and forth on which representation to use and I guess I went with the worst of both worlds.